### ISSUE TITLE
Long operation

### ISSUE THREAD

**Issue by yhl48** [microsoft/pylance-release#2098]
## Environment data

-   Language Server version: 2021.11.2
-   OS and version: Ubuntu 18.04
-   Python version (& distribution if applicable, e.g. Anaconda): Anaconda Python 3.9.7

## Expected behaviour

Pylance should work seamlessly with the code, and it is the behaviour with all but one folder.

## Actual behaviour

Takes forever to load, colour syntax only appears after minutes of waiting.

## Logs

```
[Info  - 1:37:19 PM] Pylance language server 2021.11.2 (pyright b27108d5) starting
[Info  - 1:37:19 PM] Server root directory: /home/ubuntu/.vscode-server/extensions/ms-python.vscode-pylance-2021.11.2/dist
[Info  - 1:37:19 PM] No configuration file found.
[Info  - 1:37:19 PM] No pyproject.toml file found.
[Info  - 1:37:19 PM] Setting pythonPath for service "xitorch": "/home/ubuntu/anaconda3/envs/pinns/bin/python"
[Warn  - 1:37:19 PM] stubPath /home/ubuntu/xitorch/typings is not a valid directory.
[Info  - 1:37:19 PM] Assuming Python version 3.9
[Info  - 1:37:19 PM] Assuming Python platform Linux
[Warn  - 1:37:20 PM] Exception received when installing file system watcher: TypeError [ERR_FEATURE_UNAVAILABLE_ON_PLATFORM]: The feature watch recursively is unavailable on the current platform, which is being used to run Node.js
[Warn  - 1:37:20 PM] Exception received when installing file system watcher: TypeError [ERR_FEATURE_UNAVAILABLE_ON_PLATFORM]: The feature watch recursively is unavailable on the current platform, which is being used to run Node.js
[Warn  - 1:37:20 PM] Exception received when installing file system watcher: TypeError [ERR_FEATURE_UNAVAILABLE_ON_PLATFORM]: The feature watch recursively is unavailable on the current platform, which is being used to run Node.js
[Info  - 1:37:20 PM] Searching for source files
[Info  - 1:37:20 PM] Found 80 source files
[Info  - 1:37:20 PM] Background analysis(1) root directory: /home/ubuntu/.vscode-server/extensions/ms-python.vscode-pylance-2021.11.2/dist
[Info  - 1:37:20 PM] Background analysis(1) started
[Info  - 1:39:21 PM] [BG(1)] Long operation: getSemanticTokens full at /home/ubuntu/xitorch/xitorch/_impls/linalg/symeig.py (119468ms)
[Info  - 1:39:54 PM] [BG(1)] Long operation: getSemanticTokens delta previousResultId:1637588242368 at /home/ubuntu/xitorch/xitorch/_impls/linalg/symeig.py (27103ms)
[Info  - 1:40:36 PM] [BG(1)] Long operation: checking: /home/ubuntu/xitorch/xitorch/_impls/linalg/symeig.py (42002ms)
[Info  - 1:40:36 PM] [BG(1)] Long operation: analyzing: /home/ubuntu/xitorch/xitorch/_impls/linalg/symeig.py (42017ms)
[Info  - 1:42:34 PM] [BG(1)] Long operation: getSemanticTokens delta previousResultId:1637588367375 at /home/ubuntu/xitorch/xitorch/_impls/linalg/symeig.py (118215ms)
[Info  - 1:43:36 PM] [BG(1)] Long operation: checking: /home/ubuntu/xitorch/xitorch/_impls/linalg/symeig.py (61797ms)
[Info  - 1:43:36 PM] [BG(1)] Long operation: analyzing: /home/ubuntu/xitorch/xitorch/_impls/linalg/symeig.py (61809ms)

```

## Code Snippet / Additional information

The problem seems to occur only after the update to the said version of Pylance, and the same problem is not found in other folders.

**Comment by erictraut** [comment]
To diagnose, we'll need to isolate the code that is causing the slow behavior within the type analyzer. Is this code publicly available? If so, please point us to the repo. If it's not, we'll need your help to isolate the specific module and (if possible) function.

**Comment by yhl48** [comment]
> To diagnose, we'll need to isolate the code that is causing the slow behavior within the type analyzer. Is this code publicly available? If so, please point us to the repo. If it's not, we'll need your help to isolate the specific module and (if possible) function.

Thanks for the reply, the code is publicly available here
https://github.com/yhl48/xitorch/blob/6a535d444a9955338d20784bbe31658e0c132a65/xitorch/_impls/linalg/symeig.py#L234

**Comment by erictraut** [comment]
Thanks, it makes it much easier when the code is publicly available!

Unfortunately, I'm not able to repro the behavior you're seeing. For me, pylance is able to analyze the files in this repo very quickly, in tens of milliseconds. In fact, if I change the `python.analysis.diagnosticMode` to `workspace`, it's able to analyze all 80 or so files in this repo in under 2 seconds.

I don't see anything else in your logs that would explain the behavior you're seeing.

So, we have a mystery that we need to solve. Perhaps other pylance contributors can try to repro the problem? @judej, @bschnurr, @heejaechang?

Are you seeing this behavior only for the file "symeig.py"? Or are other files affected as well?

**Comment by yhl48** [comment]
Other files are affected as well, one thing that maybe I should have mentioned, but am unsure whether or not it is related to this strange behaviour at all, is that I am connecting remotely to AWS.

**Comment by yhl48** [comment]
The problem seems to stem from ``torch.linalg.eigh`` which is called in this function https://github.com/yhl48/xitorch/blob/6a535d444a9955338d20784bbe31658e0c132a65/xitorch/_impls/linalg/symeig.py#L633. Pylance works well before the method is called, but not after.

Issue #590 seems to be relevant.

**Comment by heejaechang** [comment]
@yhl48 I can't repro this. what version of torch are you using? can you check whether this is still repro?

**Comment by judej** [comment]
Closing old issue. If this is still a problem, please reopen with the information requested. thanks

**Comment by n-gao** [comment]
@judej  I encounter the same issue in a project using [JAX](https://github.com/google/jax). The project is available here: 
https://github.com/n-gao/pesnet
Here's the pylance log:
```
[Info  - 10:35:52 AM] (2403834) Pylance language server 2022.10.10 (pyright 59bdc42e) starting
[Info  - 10:35:52 AM] (2403834) Server root directory: /nfs/homedirs/gaoni/.vscode-server/extensions/ms-python.vscode-pylance-2022.10.10/dist
[Info  - 10:35:52 AM] (2403834) Starting service instance "pesnet"
[Info  - 10:35:52 AM] (2403834) Notebook support: LSP
[Info  - 10:35:52 AM] (2403834) Interactive window support: LSP
[Info  - 10:35:52 AM] (2403834) No configuration file found.
[Info  - 10:35:52 AM] (2403834) No pyproject.toml file found.
[Info  - 10:35:52 AM] (2403834) Setting pythonPath for service "pesnet": "/nfs/staff-ssd/gaoni/miniconda3/envs/pesnet/bin/python"
[Warn  - 10:35:52 AM] (2403834) stubPath /nfs/homedirs/gaoni/repos/pesnet/typings is not a valid directory.
[Info  - 10:35:52 AM] (2403834) Assuming Python version 3.8
[Info  - 10:35:52 AM] (2403834) Assuming Python platform Linux
[Info  - 10:35:52 AM] (2403834) Searching for source files
[Info  - 10:35:56 AM] (2403834) Found 86 source files
[Info  - 10:35:56 AM] (2403834) Background analysis(1) root directory: /nfs/homedirs/gaoni/.vscode-server/extensions/ms-python.vscode-pylance-2022.10.10/dist
[Info  - 10:35:56 AM] (2403834) Background analysis(1) started
[Info  - 10:36:05 AM] (2403834) [BG(1)] Long operation: checking: /nfs/homedirs/gaoni/repos/pesnet/train.py (8252ms)
[Info  - 10:36:05 AM] (2403834) [BG(1)] Long operation: analyzing: /nfs/homedirs/gaoni/repos/pesnet/train.py (8637ms)
[Info  - 10:36:12 AM] (2403834) [BG(1)] Long operation: checking: /nfs/homedirs/gaoni/repos/pesnet/train.py (2195ms)
[Info  - 10:36:12 AM] (2403834) [BG(1)] Long operation: analyzing: /nfs/homedirs/gaoni/repos/pesnet/train.py (2213ms)
```
Here's my conda environment:
```
# Name                    Version                   Build  Channel
_libgcc_mutex             0.1                        main  
_openmp_mutex             4.5                       1_gnu  
absl-py                   1.0.0                    pypi_0    pypi
anyio                     3.5.0                    pypi_0    pypi
argon2-cffi               21.3.0                   pypi_0    pypi
argon2-cffi-bindings      21.2.0                   pypi_0    pypi
asttokens                 2.0.5                    pypi_0    pypi
astunparse                1.6.3                    pypi_0    pypi
attrs                     21.4.0                   pypi_0    pypi
autopep8                  1.6.0              pyhd3eb1b0_1  
babel                     2.9.1                    pypi_0    pypi
backcall                  0.2.0                    pypi_0    pypi
beautifulsoup4            4.10.0                   pypi_0    pypi
bleach                    4.1.0                    pypi_0    pypi
ca-certificates           2022.3.29            h06a4308_1  
cachetools                5.0.0                    pypi_0    pypi
certifi                   2021.10.8        py38h06a4308_2  
cffi                      1.15.0                   pypi_0    pypi
charset-normalizer        2.0.12                   pypi_0    pypi
chex                      0.1.1                    pypi_0    pypi
colorama                  0.4.4                    pypi_0    pypi
cudatoolkit               11.3.1               h2bc3f7f_2  
cudatoolkit-dev           11.3.1           py38h497a2fe_0    conda-forge
cudnn                     8.2.1.32             h86fa8c9_0    conda-forge
cycler                    0.11.0                   pypi_0    pypi
debugpy                   1.5.1                    pypi_0    pypi
decorator                 5.1.1                    pypi_0    pypi
defusedxml                0.7.1                    pypi_0    pypi
dm-tree                   0.1.6                    pypi_0    pypi
docopt                    0.6.2                    pypi_0    pypi
entrypoints               0.4                      pypi_0    pypi
executing                 0.8.3                    pypi_0    pypi
flatbuffers               2.0                      pypi_0    pypi
flax                      0.4.0                    pypi_0    pypi
fonttools                 4.29.1                   pypi_0    pypi
future                    0.18.2                   pypi_0    pypi
gast                      0.5.3                    pypi_0    pypi
gitdb                     4.0.9                    pypi_0    pypi
gitpython                 3.1.27                   pypi_0    pypi
google-auth               2.6.0                    pypi_0    pypi
google-auth-oauthlib      0.4.6                    pypi_0    pypi
google-pasta              0.2.0                    pypi_0    pypi
grpcio                    1.44.0                   pypi_0    pypi
h5py                      3.1.0                    pypi_0    pypi
idna                      3.3                      pypi_0    pypi
impi-rt                   2021.5.1                 pypi_0    pypi
importlib-metadata        4.11.2                   pypi_0    pypi
importlib-resources       5.4.0                    pypi_0    pypi
intel-cmplr-lib-rt        2022.0.2                 pypi_0    pypi
intel-cmplr-lic-rt        2022.0.2                 pypi_0    pypi
intel-fortran-rt          2022.0.2                 pypi_0    pypi
intel-openmp              2022.0.2                 pypi_0    pypi
ipykernel                 6.9.2                    pypi_0    pypi
ipympl                    0.9.0                    pypi_0    pypi
ipython                   8.1.0                    pypi_0    pypi
ipython-genutils          0.2.0                    pypi_0    pypi
ipywidgets                7.7.0                    pypi_0    pypi
jax                       0.3.1                    pypi_0    pypi
jaxboard                  0.0.1                    pypi_0    pypi
jaxlib                    0.3.0+cuda11.cudnn82          pypi_0    pypi
jaxopt                    0.5                      pypi_0    pypi
jedi                      0.18.1                   pypi_0    pypi
jinja2                    3.0.3                    pypi_0    pypi
jraph                     0.0.2.dev0               pypi_0    pypi
json5                     0.9.6                    pypi_0    pypi
jsonpickle                1.5.2                    pypi_0    pypi
jsonschema                4.4.0                    pypi_0    pypi
jupyter-client            7.1.2                    pypi_0    pypi
jupyter-core              4.9.2                    pypi_0    pypi
jupyter-server            1.15.6                   pypi_0    pypi
jupyterlab                3.3.2                    pypi_0    pypi
jupyterlab-pygments       0.1.2                    pypi_0    pypi
jupyterlab-server         2.11.2                   pypi_0    pypi
jupyterlab-widgets        1.1.0                    pypi_0    pypi
keras                     2.8.0                    pypi_0    pypi
keras-preprocessing       1.1.2                    pypi_0    pypi
kiwisolver                1.3.2                    pypi_0    pypi
ld_impl_linux-64          2.35.1               h7274673_9  
libclang                  13.0.0                   pypi_0    pypi
libffi                    3.3                  he6710b0_2  
libgcc-ng                 9.3.0               h5101ec6_17  
libgomp                   9.3.0               h5101ec6_17  
libstdcxx-ng              9.3.0               hd4cf53a_17  
markdown                  3.3.6                    pypi_0    pypi
markupsafe                2.1.1                    pypi_0    pypi
matplotlib                3.5.1                    pypi_0    pypi
matplotlib-inline         0.1.3                    pypi_0    pypi
merge-args                0.1.4                    pypi_0    pypi
mistune                   0.8.4                    pypi_0    pypi
mpmath                    1.2.1                    pypi_0    pypi
msgpack                   1.0.3                    pypi_0    pypi
munch                     2.5.0                    pypi_0    pypi
nbclassic                 0.3.7                    pypi_0    pypi
nbclient                  0.5.13                   pypi_0    pypi
nbconvert                 6.4.4                    pypi_0    pypi
nbformat                  5.2.0                    pypi_0    pypi
ncurses                   6.3                  h7f8727e_2  
nest-asyncio              1.5.4                    pypi_0    pypi
notebook                  6.4.10                   pypi_0    pypi
notebook-shim             0.1.0                    pypi_0    pypi
numpy                     1.22.2                   pypi_0    pypi
oauthlib                  3.2.0                    pypi_0    pypi
openssl                   1.1.1n               h7f8727e_0  
opt-einsum                3.3.0                    pypi_0    pypi
optax                     0.0.9                    pypi_0    pypi
packaging                 21.3                     pypi_0    pypi
pandas                    1.4.1                    pypi_0    pypi
pandocfilters             1.5.0                    pypi_0    pypi
parso                     0.8.3                    pypi_0    pypi
pesnet                    0.1.0                     dev_0    <develop>
pexpect                   4.8.0                    pypi_0    pypi
pickleshare               0.7.5                    pypi_0    pypi
pillow                    9.0.1                    pypi_0    pypi
pip                       21.2.4           py38h06a4308_0  
prometheus-client         0.13.1                   pypi_0    pypi
prompt-toolkit            3.0.28                   pypi_0    pypi
protobuf                  3.19.4                   pypi_0    pypi
psutil                    5.9.0                    pypi_0    pypi
ptyprocess                0.7.0                    pypi_0    pypi
pure-eval                 0.2.2                    pypi_0    pypi
py-cpuinfo                8.0.0                    pypi_0    pypi
pyasn1                    0.4.8                    pypi_0    pypi
pyasn1-modules            0.2.8                    pypi_0    pypi
pycodestyle               2.8.0              pyhd3eb1b0_0  
pycparser                 2.21                     pypi_0    pypi
pygments                  2.11.2                   pypi_0    pypi
pymongo                   4.0.1                    pypi_0    pypi
pyparsing                 3.0.7                    pypi_0    pypi
pyrsistent                0.18.1                   pypi_0    pypi
pyscf                     2.0.0a0                  pypi_0    pypi
pyscf-properties          0.1.0                    pypi_0    pypi
pyscfad                   0.1.0                    pypi_0    pypi
python                    3.8.12               h12debd9_0  
python-dateutil           2.8.2                    pypi_0    pypi
python-version            0.0.2                    pypi_0    pypi
python_abi                3.8                      2_cp38    conda-forge
pytz                      2021.3                   pypi_0    pypi
pyyaml                    6.0                      pypi_0    pypi
pyzmq                     22.3.0                   pypi_0    pypi
readline                  8.1.2                h7f8727e_1  
requests                  2.27.1                   pypi_0    pypi
requests-oauthlib         1.3.1                    pypi_0    pypi
rsa                       4.8                      pypi_0    pypi
sacred                    0.8.2                    pypi_0    pypi
scipy                     1.8.0                    pypi_0    pypi
seml                      0.3.6                    pypi_0    pypi
seml-logger               0.1.0                     dev_0    <develop>
send2trash                1.8.0                    pypi_0    pypi
setuptools                58.0.4           py38h06a4308_0  
six                       1.16.0                   pypi_0    pypi
smmap                     5.0.0                    pypi_0    pypi
sniffio                   1.2.0                    pypi_0    pypi
soupsieve                 2.3.1                    pypi_0    pypi
sqlite                    3.37.2               hc218d9a_0  
stack-data                0.2.0                    pypi_0    pypi
sympy                     1.9                      pypi_0    pypi
tabulate                  0.8.10                   pypi_0    pypi
tensorboard               2.10.0                   pypi_0    pypi
tensorboard-data-server   0.6.1                    pypi_0    pypi
tensorboard-plugin-wit    1.8.1                    pypi_0    pypi
tensorboardx              2.5                      pypi_0    pypi
tensorflow                2.8.0                    pypi_0    pypi
tensorflow-gpu            2.8.0                    pypi_0    pypi
tensorflow-io-gcs-filesystem 0.24.0                   pypi_0    pypi
termcolor                 1.1.0                    pypi_0    pypi
terminado                 0.13.3                   pypi_0    pypi
testpath                  0.6.0                    pypi_0    pypi
tf-estimator-nightly      2.8.0.dev2021122109          pypi_0    pypi
tk                        8.6.11               h1ccaba5_0  
toml                      0.10.2             pyhd3eb1b0_0  
toolz                     0.11.2                   pypi_0    pypi
tornado                   6.1                      pypi_0    pypi
tqdm                      4.63.0                   pypi_0    pypi
traitlets                 5.1.1                    pypi_0    pypi
typing-extensions         4.1.1                    pypi_0    pypi
uncertainties             3.1.6                    pypi_0    pypi
urllib3                   1.26.8                   pypi_0    pypi
wcwidth                   0.2.5                    pypi_0    pypi
webencodings              0.5.1                    pypi_0    pypi
websocket-client          1.3.1                    pypi_0    pypi
werkzeug                  2.0.3                    pypi_0    pypi
wheel                     0.37.1             pyhd3eb1b0_0  
widgetsnbextension        3.6.0                    pypi_0    pypi
wrapt                     1.13.3                   pypi_0    pypi
xz                        5.2.5                h7b6447c_0  
zipp                      3.7.0                    pypi_0    pypi
zlib                      1.2.11               h7f8727e_4
```
What might be related: I develop on a server via SSH where the directory is mounted via nfs. Though, the network interface shouldn't be a bottleneck nor the compute of the server.

**Comment by debonte** [comment]
> I encounter the same issue in a project using [JAX](https://github.com/google/jax).

@n-gao, could you please open a new issue for this?

**Comment by dobos** [comment]
I'm seeing this issue as well and it's particularly slow to analyze jupyter notebooks. The main problem is that it blocks typing while the analysis is running in the background which makes work very disruptive. I think one obvious solution would be to run the analysis async, and not on the thread that updates the text in the editor. If it takes longer than a certain limit, than no quick info or IntelliSense info is displayed and that's it. If I need the info I'll wait but I don't want to wait for my typing to appear in the editor.

**Comment by rchiodo** [comment]
> I'm seeing this issue as well and it's particularly slow to analyze jupyter notebooks. The main problem is that it blocks typing while the analysis is running in the background which makes work very disruptive. I think one obvious solution would be to run the analysis async, and not on the thread that updates the text in the editor. If it takes longer than a certain limit, than no quick info or IntelliSense info is displayed and that's it. If I need the info I'll wait but I don't want to wait for my typing to appear in the editor.

Blocking typing while analysis is running shouldn't happen in VS code. Can you open a new issue? Completions don't finish but you should still be able to type. VS code is supposed to cancel the completion request when you type another letter.

**Comment by dobos** [comment]
> Blocking typing while analysis is running shouldn't happen in VS code. Can you open a new issue? Completions don't finish but you should still be able to type. VS code is supposed to cancel the completion request when you type another letter.

It's happening in jupyter notebooks. I'll open a new issue but can't promise to put together a reproduction. I have a huge workspace with a pretty long jupyter notebook that I can't make public right now.

**Comment by saipavanc** [comment]
I have been experiencing this issue consistently since a few weeks now, any of the new updates do not seem to work. I get the following message in my output:
```
2024-02-28 12:39:12.176 [info] [Info  - 12:39:12 PM] (4096) [BG(7)] Long operation: checking: file:///Test.ipynb.py#Y235sZmlsZQ%3D%3D (3812ms)
2024-02-28 12:39:12.176 [info] [Info  - 12:39:12 PM] (4096) [BG(7)] Long operation: analyzing: file:///Test.ipynb.py#Y235sZmlsZQ%3D%3D (3812ms)
2024-02-28 12:39:12.176 [info] [Info  - 12:39:12 PM] (4096) [BG(7)] Long operation: analyzing: file:///Test.ipynb.py#pylancePrefixCell (3833ms)
2024-02-28 12:39:19.848 [info] [Info  - 12:39:19 PM] (4096) [BG(7)] Long operation: getSemanticTokens full at file:///Test.ipynb.py#Y104sZmlsZQ%3D%3D (7659ms)
2024-02-28 12:39:29.655 [info] [Info  - 12:39:29 PM] (4096) [BG(7)] Long operation: checking: file:///Test.ipynb.py#Y235sZmlsZQ%3D%3D (9588ms)
2024-02-28 12:39:29.655 [info] [Info  - 12:39:29 PM] (4096) [BG(7)] Long operation: analyzing: file:///Test.ipynb.py#Y235sZmlsZQ%3D%3D (9588ms)
2024-02-28 12:39:30.817 [info] [Info  - 12:39:30 PM] (4096) [BG(7)] Long operation: analyzing: file:///Test.ipynb.py#pylancePrefixCell (10805ms)
2024-02-28 12:40:17.408 [info] [Info  - 12:40:17 PM] (4096) [BG(7)] Long operation: checking: file:///Test.ipynb.py#Y235sZmlsZQ%3D%3D (9711ms)
2024-02-28 12:40:17.408 [info] [Info  - 12:40:17 PM] (4096) [BG(7)] Long operation: analyzing: file:///Test.ipynb.py#Y235sZmlsZQ%3D%3D (9712ms)
2024-02-28 12:40:18.457 [info] [Info  - 12:40:18 PM] (4096) [BG(7)] Long operation: analyzing: file:///Test.ipynb.py#pylancePrefixCell (10822ms)
2024-02-28 12:40:56.366 [info] [Info  - 12:40:56 PM] (4096) [BG(7)] Long operation: checking: file:///Test.ipynb.py#Y235sZmlsZQ%3D%3D (9301ms)
2024-02-28 12:40:56.366 [info] [Info  - 12:40:56 PM] (4096) [BG(7)] Long operation: analyzing: file:///Test.ipynb.py#Y235sZmlsZQ%3D%3D (9301ms)
2024-02-28 12:40:57.428 [info] [Info  - 12:40:57 PM] (4096) [BG(7)] Long operation: analyzing: file:///Test.ipynb.py#pylancePrefixCell (10386ms)
```
While this is happening, I have no prompts what so ever from vscode, about the object attributes or methods. Is there any other info that I can generate which might be helpful to fix this?

**Comment by rchiodo** [comment]
> I have been experiencing this issue consistently since a few weeks now, any of the new updates do not seem to work. I get the following message in my output:
> 
> ```
> 2024-02-28 12:39:12.176 [info] [Info  - 12:39:12 PM] (4096) [BG(7)] Long operation: checking: file:///Test.ipynb.py#Y235sZmlsZQ%3D%3D (3812ms)
> 2024-02-28 12:39:12.176 [info] [Info  - 12:39:12 PM] (4096) [BG(7)] Long operation: analyzing: file:///Test.ipynb.py#Y235sZmlsZQ%3D%3D (3812ms)
> 2024-02-28 12:39:12.176 [info] [Info  - 12:39:12 PM] (4096) [BG(7)] Long operation: analyzing: file:///Test.ipynb.py#pylancePrefixCell (3833ms)
> 2024-02-28 12:39:19.848 [info] [Info  - 12:39:19 PM] (4096) [BG(7)] Long operation: getSemanticTokens full at file:///Test.ipynb.py#Y104sZmlsZQ%3D%3D (7659ms)
> 2024-02-28 12:39:29.655 [info] [Info  - 12:39:29 PM] (4096) [BG(7)] Long operation: checking: file:///Test.ipynb.py#Y235sZmlsZQ%3D%3D (9588ms)
> 2024-02-28 12:39:29.655 [info] [Info  - 12:39:29 PM] (4096) [BG(7)] Long operation: analyzing: file:///Test.ipynb.py#Y235sZmlsZQ%3D%3D (9588ms)
> 2024-02-28 12:39:30.817 [info] [Info  - 12:39:30 PM] (4096) [BG(7)] Long operation: analyzing: file:///Test.ipynb.py#pylancePrefixCell (10805ms)
> 2024-02-28 12:40:17.408 [info] [Info  - 12:40:17 PM] (4096) [BG(7)] Long operation: checking: file:///Test.ipynb.py#Y235sZmlsZQ%3D%3D (9711ms)
> 2024-02-28 12:40:17.408 [info] [Info  - 12:40:17 PM] (4096) [BG(7)] Long operation: analyzing: file:///Test.ipynb.py#Y235sZmlsZQ%3D%3D (9712ms)
> 2024-02-28 12:40:18.457 [info] [Info  - 12:40:18 PM] (4096) [BG(7)] Long operation: analyzing: file:///Test.ipynb.py#pylancePrefixCell (10822ms)
> 2024-02-28 12:40:56.366 [info] [Info  - 12:40:56 PM] (4096) [BG(7)] Long operation: checking: file:///Test.ipynb.py#Y235sZmlsZQ%3D%3D (9301ms)
> 2024-02-28 12:40:56.366 [info] [Info  - 12:40:56 PM] (4096) [BG(7)] Long operation: analyzing: file:///Test.ipynb.py#Y235sZmlsZQ%3D%3D (9301ms)
> 2024-02-28 12:40:57.428 [info] [Info  - 12:40:57 PM] (4096) [BG(7)] Long operation: analyzing: file:///Test.ipynb.py#pylancePrefixCell (10386ms)
> ```
> 
> While this is happening, I have no prompts what so ever from vscode, about the object attributes or methods. Is there any other info that I can generate which might be helpful to fix this?
> 
> Software details: Vscode: 1.86.2 pylance: 2024.2.2 Os: Arch linux 6.7.4-arch1-1

We'd need the code in the notebook. Please open a new issue.

**Comment by rchiodo** [comment]
I believe @saipanvc's issue is this one:
https://github.com/microsoft/pyright/issues/7159

**Comment by saipavanc** [comment]
Can I ask what makes this different from prompts in Jupyter lab for example? I ran the same notebook in Jupyter Lab, and there is no problem for prompts and no lag associated with it as well. Is the process fundamentally different between pylance and whatever's used in Jupyter Lab?

**Comment by rchiodo** [comment]
> Is the process fundamentally different between pylance and whatever's used in Jupyter Lab?

Jupyter lab does not use pylance (or pyright) and is also running the python code itself. The python runtime has all of the types for everything already in memory, well because it's actually loading the sympy modules. Pylance is a static analyzer. We have to try and guess what will happen at runtime and this guessing can take a long time when the code has no typing information. Sympy has very little types embedded in it.
